config/lib/seed.js

```diff
@@ -92,3 +65,85 @@ function reportError(err) {
   console.log(err);
   console.log();
 }
+
+function performSeedUser(user) {
+
+  var User = mongoose.model('User');
+
+  return new Promise(function(resolve, reject) {
+
+    var seedUser = {
+      username: 'user',
+      password: 'User_Password1!',
+      provider: 'local',
+      email: 'user@localhost.com',
+      firstName: 'User',
+      lastName: 'Local',
+      displayName: 'User Local',
+      roles: ['user']
+    };
+
+    var seedAdmin = {
+      username: 'admin',
+      password: 'Admin_Password1!',
+      provider: 'local',
+      email: 'admin@localhost.com',
+      firstName: 'Admin',
+      lastName: 'Local',
+      displayName: 'Admin Local',
+      roles: ['user', 'admin']
+    };
+
+    var user = null;
+
+    //If production only seed admin if it does not exist
+    if (process.env.NODE_ENV === 'production') {
+      //Add Local Admin
+
+      user = new User(seedAdmin);
+
+      User.generateRandomPassphrase()
+      .then(function(password) {
+        return saveUser(user, password);
+      })
+      .then(function() {
+          resolve();
+        })
+        .catch(reportError);
+
+    } else {
+
+      //Add Local User
+      user = new User(seedUser);
+
+      User.generateRandomPassphrase()
+      .then(function(password) {
+        return saveUser(user, password);
+      })
+      .then(function() {
+        var user = new User(seedAdmin);
+
+        User.generateRandomPassphrase()
+        .then(function(password) {
+          return saveUser(user, password);
+        })
+        .then(function() {
+          resolve();
+        })
+        .catch(reportError);
+      })
+      .catch(reportError);
+
+    }
+
+  });
+
+}
+
+module.exports.start = function start() {
+
+  return new Promise(function(resolve, reject) {
+    resolve(performSeedUser());
```

Comment:
Why not just have the start return a resolve of the function? This seems too complicated.